mimic/rest/rackconnect_v3_api.py

```diff
@@ -0,0 +1,436 @@
+
+    :ivar int default_lbs: The number of default load balancers to be created
+        per tenant per region
+    """
+
+    def __init__(self, regions=("ORD",), default_lbs=1):
+        """
+        Construct a :class:`RackConnectV3` object.
+        """
+        self.regions = regions
+        self.default_lbs = default_lbs
+
+    def catalog_entries(self, tenant_id):
+        """
+        Catalog entry for RackConnect V3 endpoints.
+        """
+        return [
+            Entry(tenant_id, "rax:rackconnect", "rackconnect", [
+                Endpoint(tenant_id, region, text_type(uuid4()), prefix="v3")
+                for region in self.regions
+            ])
+        ]
+
+    def resource_for_region(self, region, uri_prefix, session_store):
+        """
+        Return an IResource implementing a public RackConnect V3 region
+        endpoint.
+        """
+        return RackConnectV3Region(
+            iapi=self,
+            uri_prefix=uri_prefix,
+            session_store=session_store,
+            region_name=region,
+            default_lbs=self.default_lbs).app.resource()
+
+
+lb_pool_attrs = [
+    Attribute("id", default_factory=lambda: text_type(uuid4())),
+    Attribute("name", default_value=u"default", instance_of=text_type),
+    Attribute("port", default_value=80, instance_of=int),
+    Attribute("status", default_value=u"ACTIVE", instance_of=text_type),
+    Attribute("status_detail", default_value=None),
+    Attribute("virtual_ip", default_factory=random_ipv4),
+    Attribute('nodes', default_factory=list, instance_of=list)]
+
+
+@attributes(lb_pool_attrs, apply_with_cmp=False)
+class LoadBalancerPool(object):
+    """
+    Represents a Load Balancer Pool, which cannot be created via the API.
+
+    :param str id: The ID of the load balancer pool - is randomly generated
+        if not provided
+    :param str name: The name of the load balancer pool - defaults to
+        'default'
+    :param int port: The incoming port of the load balancer - defaults to 80
+    :param str status: The status of the load balancer - defaults to "ACTIVE"
+    :param str status_detail: Any details about the status - defaults to None
+    :param list nodes: :class:`LoadBalancerPoolNode`s
+    """
+    def as_json(self):
+        """
+        Create a JSON-serializable representation of the contents of this
+        object, which can be used in a REST response for a request for the
+        details of this particular object
+        """
+        # no dictionary comprehensions in py2.6
+        response = dict([(attr, getattr(self, attr)) for attr in
+                         attribute_names(lb_pool_attrs) if attr != "nodes"])
+        response['node_counts'] = {
+            "cloud_servers": len(self.nodes),
+            "external": 0,
+            "total": len(self.nodes)
+        }
+        return response
+
+    def node_by_cloud_server(self, cloud_server_id):
+        """
+        Find a node by it's cloud server ID.
```

Comment:
I don't know if the default return value vs an exception should be documented here. I'm inclined to say yes.